core/node/eth_sender/src/eth_fees_oracle.rs

```diff
@@ -54,6 +57,16 @@ impl GasAdjusterFeesOracle {
         self.assert_fee_is_not_zero(blob_base_fee_per_gas, "blob");
         let blob_base_fee_per_gas = Some(blob_base_fee_per_gas);
 
+        if base_fee_per_gas > self.max_acceptable_base_fee_in_wei {
```

Comment:
AFAIU, the `EthFees.base_fee_per_gas` returned from the method can still be greater than `self.max_acceptable_base_fee_in_wei` if there's a previously sent tx. Is this intended? Ditto for `calculate_fees_no_blob_sidecar()`.

Reply:
Yeah this is intended. 

The goal is just to avoid transactions when the network base fee exceeds some threshold (i.e when someone is spamming L1)

It is okay to exceed `max_acceptable_base_fee_in_wei` for a few transactions if it is to solve the `replacement transaction underpriced issue`

Reply:
On this topic, we sometime facing issue of transactions being stucked because previous sent tx exceed the rpc provider limit and the current logic does not allow new transactions to have lower fee to avoid `replacement transaction underpriced issue`

In that case, we need to manually clean the db which is a risky operation.

We probably to design a tool or have an auto-cleaning mechanism in place, but this is a completely different problem unrelated to this PR

Reply:
This is a very clever way to implement such cap, I really like it